frontend/lib/parsing/parser-error-classes-list.cpp

```diff
@@ -418,6 +418,33 @@ void ErrorIllegalUseImport::write(ErrorWriterBase& wr) const {
   wr.note(clause, "only identifiers and 'dot' expressions are supported");
 }
 
+void ErrorInvalidGpuAssertion::write(ErrorWriterBase& wr) const {
+  auto loop = std::get<const uast::Loop*>(info);
+  // auto attr = std::get<const uast::Attribute*>(info);
+
+  const char* loopTypes = nullptr;
+  if (loop->isFor()) {
+    loopTypes = "for";
+  } else if (loop->isWhile()) {
+    loopTypes = "while";
+  } else if (loop->isDoWhile()) {
+    loopTypes = "do-while";
+  }
```

Comment:
I made this error more general ("statement does not support the attribute") in case it's not a loop. I think that's the appropriate way to handle this, and results in "reasonable" output in other cases.

Reply:
Then should you change `if (loop)` to `if (loop && loopTypes)`?

Reply:
It should be just `loopTypes` -- I have it fixed on my branch, and will push in a moment.